llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -11850,12 +11850,11 @@ void SelectionDAGISel::LowerArguments(const Function &F) {
 
     FPClassTest NoFPClass = Arg.getNoFPClass();
     if (NoFPClass != fcNone) {
-      EVT I64EVT = EVT::getIntegerVT(*DAG.getContext(), 64);
       SDValue SDNoFPClass =
-          DAG.getConstant(static_cast<uint64_t>(NoFPClass), dl, I64EVT);
-      SDNodeFlags ResFlags = Res->getFlags();
+          DAG.getTargetConstant(static_cast<uint64_t>(NoFPClass), dl,
+                                TLI->getPointerTy(DAG.getDataLayout()));
```

Comment:
This should not use getPointerTy. I would make it an unconditional i32

Reply:
Maybe a good idea, since no 16bit target is supported by LLVM yet, while I wish that we can keep the possibility.

Reply:
The type of target constants does not matter at all. Making this a variable type is added complexity for no benefit